### ISSUE TITLE
generation failed if github commit api return `404 Not Found`

### ISSUE THREAD

**Issue by nantas** [skywinder/Github-Changelog-Generator#57]
Here's what I've got when running generator for my private repo:
```
/Users/nantas/.rvm/gems/ruby-2.1.5/gems/github_api-0.12.3/lib/github_api/response/raise_error.rb:14:in `on_complete': GET https://api.github.com/repos/fireball-x/dev/git/commits/38f8efae9be54183fa08373ed203d7c955917003?access_token=xxx: 404 Not Found (Github::Error::NotFound)
```

For some reason commit `38f8efae9be54` cannot be found in the commit history, but it's still referenced as the commit that closed an issue.

I'm not sure how important it is to get the commit date as the issue closed date. But if the commit is not found, the generator should just use `actual_date` of the issue without blocking the whole process. Is that right?

**Comment by skywinder** [comment]
@nantas yes, that's absolutely right! Thank you!
Can you tell me, which repo generate this issue?

**Comment by nantas** [comment]
I did more tests and find out the exact cause of the issue.
You can reproduce the issue with my test repo: https://github.com/nantas/changelog_test

With the test repo, the commit that return `404 Not Found` is a commit in another repo:  https://github.com/nantas/changelog_test2/commit/34a8eee7aa272d5226126d986764a7945a9253b9

The program assumes that all commits that close issues are from the same repo. But in our case and many other application consists of multiple submodules that's not the case. We use one main repo for managing the project and we close main repo's issue from submodule repos.

I think an improvement on that matter will do great help to developers like my team. And also a fallback solution for github api error return would be great too.

**Comment by skywinder** [comment]
@nantas Thanks a lot to clarifying! I will handle it soon, get in touch!

**Comment by skywinder** [comment]
@nantas Thanks a lot. Issue fixed in `1.3.2`.
I take some research, but still don't know, how to fetch repo address from events [github api](https://developer.github.com/v3/activity/events/). It's needed to define correct closed date.
Now I use just closed date of this issue (but sometimes this date is not, that we looking for. see #37 for more details).
Let me know, if you find methods to fetch commit's repo address from github api.
